devtools/datasette/publish.py

```diff
@@ -82,9 +83,78 @@ def inspect_data(datasets: list[str], pudl_out: Path) -> str:
     return inspect_output
 
 
-def metadata(pudl_out: Path) -> str:
+def check_tables_have_metadata(
+    metadata: DatasetteMetadata,
+    pudl_out: Path,
+    datasets: list[str],
+) -> None:
+    """Check to make sure all tables in the databases have metadata.
+
+    This function fails if there are tables present in a database
+    that do not have structure metadata.
+
+    Args:
+        metadata: The structure metadata for the datasette deployment
+        pudl_out: The directory that contains the pudl outputs
+        datasets: The list of datasets to test.
+    """
+    resources = {}
+    resources |= metadata.xbrl_resources
+    resources["pudl"] = metadata.resources
+
+    resource_exceptions = {"pudl": {"alembic_version"}}
+
+    tables_missing_metadata_results = {}
+
+    for dataset in datasets:
```

Comment:
I'm having a hard time following here because I don't know what "dataset" refers to.  It seems like maybe it's actually a database, not a dataset?

Reply:
I'll update the names to make it clearer.